homeassistant/components/zwave_js/binary_sensor.py

```diff
@@ -335,58 +285,58 @@ class ZWaveNotificationBinarySensor(ZWaveBaseEntity, BinarySensorEntity):
     """Representation of a Z-Wave binary_sensor from Notification CommandClass."""
 
     def __init__(
-        self, config_entry: ConfigEntry, client: ZwaveClient, info: ZwaveDiscoveryInfo
+        self,
+        config_entry: ConfigEntry,
+        client: ZwaveClient,
+        info: ZwaveDiscoveryInfo,
+        state_key: str,
     ) -> None:
         """Initialize a ZWaveNotificationBinarySensor entity."""
         super().__init__(config_entry, client, info)
+        self.state_key = state_key
         # check if we have a custom mapping for this value
         self._mapping_info = self._get_sensor_mapping()
 
     @property
     def is_on(self) -> bool:
         """Return if the sensor is on or off."""
-        if self._mapping_info:
-            return self.info.primary_value.value in self._mapping_info["states"]
-        return bool(self.info.primary_value.value != 0)
+        return self.info.primary_value.value is int(self.state_key)
 
     @property
     def name(self) -> str:
-        """Return default name from device name and value name combination."""
-        node_name = self.info.node.name or self.info.node.device_config.description
-        property_name = self.info.primary_value.property_name
-        property_key_name = self.info.primary_value.property_key_name
-        return f"{node_name}: {property_name}: {property_key_name}"
+        """Return entity name with the state label appended."""
+        state_label = self.info.primary_value.metadata.states[self.state_key]
+        return f"{super().name} - {state_label}"
```

Comment:
this resolves to "`node name: self.info.primary_value.metadata.label - state_label`. Is that what we want? is the concern that adding `property_name` and `property_key_name` makes it too long?

Reply:
Yes, because without we can have overlap. For example see the various smoke sensors and their maintenance and test values: https://github.com/zwave-js/node-zwave-js/blob/b62505309ef31c4717d321ae0fb3da2ef41dd10a/packages/config/config/notifications.json#L99

Reply:
BTW: I like the short name better (only the value) but this is safer

Reply:
I think we will still have overlap with this change: https://pastebin.com/ZYgLTdS8 look at the values at 2476 and 2527. With this naming strategy, they would have the same name. That's why it's IMO useful to have property name in there.

Reply:
You are right. In case of these values the propertyname actually holds the right label. I'll adjust it...

Reply:
well I think you need both. It ends up being `node name: property name: property key name - state label`

Reply:
hopefully HA has a way to handle long character names

Reply:
from Discord, we decided `node_name: property_name - state_label` was sufficient for binary sensors and sensors will get `node_name: property_name - property_key_name`